worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -961,10 +961,15 @@ public int center(Region region, Pattern pattern) throws MaxChangedBlocksExcepti
         checkNotNull(pattern);
 
         Vector center = region.getCenter();
+        Vector vec1 = new Vector((int) center.getX(), (int) center.getY(), (int) center.getZ());
+        Vector vec2 = center.toBlockVector();
+        boolean condX = (region.getWidth() % 2) == 0;
+        boolean condZ = (region.getLength() % 2) == 0;
+        Vector vec3 = new Vector(vec2.getX() + (condX ? 0.5*Math.signum(vec2.getX()) : 0), vec2.getY(), vec2.getZ() + (condZ ? 0.5*Math.signum(vec2.getZ()) : 0));
```

Comment:
I'm another contributor from the fork. 
Basically, the problem was on the third argument of the CuboidRegion constructor used to define centerRegion, where the vector's coordinates were not being properly rounded. In particular, if the x or z coordinates were -a.5 (a being a positive integer) (i.e. when the width or length of the region were even), the number was rounded to -a and not to -(a+1).
We added a new commit to the pull request where we simplified this bug correction.